src/atoms/Link/Link.native.js

```diff
@@ -0,0 +1,71 @@
+import React, { useContext, useState, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import Text from '../Text';
+import styled, { ThemeContext } from 'styled-components/native';
+
+const StyledTouchable = styled.TouchableOpacity`
+  align-self: flex-start;
+`;
+
+const resolveSizeToTextProps = (linkSize) => {
```

Comment:
create an object `styles` and then keep adding functions there so you have the context tied to it.

```js
const styles = {
	size: (size) => {
		switch()
		...
	}
}
```

Reply:
I will have to do something like this
```
      <Text
        underline={active}
        color={active ? activeTextColor : inactiveTextColor}
        size={styles.size({ size })}
      >
        {children}
      </Text>
```

```
const styles = {
  size: ({ size }) => {
    switch (size) {
      case 'small':
      default:
        return 'xsmall';
      case 'medium':
        return 'medium';
      case 'large':
        return 'large';
    }
  },
};
```

since we're resolving the size and then passing it down to another Text component



Is that okay?

Reply:
Another way is this

```
const linkToTextSize = {
  small: 'xsmall',
  medium: 'medium',
  large: 'large',
};
```

```
size={linkToTextSize[size]}
```